timemachine/cpp/src/kernels/k_nonbonded_pair_list.cuh

```diff
@@ -154,8 +153,16 @@ void __global__ k_nonbonded_pair_list(
         gj_y += FLOAT_TO_FIXED_NONBONDED(-delta_prefactor * delta_y);
         gj_z += FLOAT_TO_FIXED_NONBONDED(-delta_prefactor * delta_z);
 
-        // energy is size extensive so this may not be a good idea
-        energy += FLOAT_TO_FIXED_NONBONDED(u);
+        if (u_buffer) {
+            // Don't add energy if it overflows, but always add the overflow count
+            int overflow_count = 0;
+            if (!energy_overflowed<RealType>(u, overflow_count)) {
+                // energy is size extensive so this may not be a good idea
+                unsigned long long energy = FLOAT_TO_FIXED_NONBONDED(u);
+                accumulate<Negated>(u_buffer + atom_i_idx, energy);
+            }
+            atomicAdd(u_overflow_count, Negated ? -overflow_count : overflow_count);
```

Comment:
This logic is in the PR description, but it would be useful to include a line comment (in nonbonded_common? not sure where...) that the overflow_count can either be incremented or decremented when overflows are encountered.

Reply:
Currently, it is implemented as specific to Nonbonded, but this will eventually apply to the other potentials. What about adding the documentation to the `accumulate_energies` function in the wrap kernels? It is where we implement the behavior that a non-zero overflow_count results in a nan.

Reply:
In `wrap_kernels.cpp::accumulate_energies` sounds appropriate!

Reply:
Expanded the docstring to include mention about incrementing/decrementing overflows in [9c8c748](https://github.com/proteneer/timemachine/pull/1084/commits/9c8c7488b72de7c8d1b9380b9f322512e4017999)

Reply:
It might be worth noting that the (negated version of) `NonbondedPairList` is specialized to handling exclusions. I'm not sure of other use cases, but it seems like this logic could lead to incorrect results in the case that the negated pair list isn't used together with a nonbonded potential with the same interactions (since overflows in different terms could erroneously "cancel").

Reply:
Added docstring to the NonbondedPairList being more explicit about its use case [6221fad](https://github.com/proteneer/timemachine/pull/1084/commits/6221fad25f36f3b55785db0fd570da8f6f05ec43)